solutions/1000-1099/1000-minimum-cost-to-merge-stones-hard.md

```diff
@@ -0,0 +1,322 @@
+---
+description: 'Author: @hwaiting_dk | https://leetcode.com/problems/minimum-cost-to-merge-stones/'
+tags: [Array, Dynamic Programming]
+---
+
+# 1000 - Minimum Cost to Merge Stones (Hard)
+
+## Problem Link
+
+https://leetcode.com/problems/minimum-cost-to-merge-stones/
+
+## Problem Statement
+
+There are n piles of stones arranged in a row. The ith pile has stones[i] stones.
+A move consists of merging exactly k consecutive piles into one pile, and the cost of this move is equal to the total number of stones in these k piles.
+Return the minimum cost to merge all piles of stones into one pile. If it is impossible, return -1.
+
+**Example 1:**
+```
+Input: stones = [3,2,4,1], k = 2
+Output: 20
+Explanation: We start with [3, 2, 4, 1].
+We merge [3, 2] for a cost of 5, and we are left with [5, 4, 1].
+We merge [4, 1] for a cost of 5, and we are left with [5, 5].
+We merge [5, 5] for a cost of 10, and we are left with [10].
+The total cost was 20, and this is the minimum possible.
+```
+
+**Example 2:**
+```
+Input: stones = [3,2,4,1], k = 3
+Output: -1
+Explanation: After any merge operation, there are 2 piles left, and we can't merge anymore.  
+So the task is impossible.
+```
+
+**Example 3:**
+```
+Input: stones = [3,5,1,2,6], k = 3
+Output: 25
+Explanation: We start with [3, 5, 1, 2, 6].
+We merge [5, 1, 2] for a cost of 8, and we are left with [3, 8, 6].
+We merge [3, 8, 6] for a cost of 17, and we are left with [17].
+The total cost was 25, and this is the minimum possible.
+```
+**Constraints:**
+* `n == stones.length`
+* `1 <= n <= 30`
+* `1 <= stones[i] <= 100`
+* `2 <= k <= 30`
+
+<SolutionAuthor name="@hwaiting_dk"/>
+
+# Problem Breakdown
+Approach : It's variation of MCM.
+Let's understand problem first,
+stones = [3,5,1,2,6], k = 3
+
+here we are given stone array and k piles, we have to merge k consecutive piles into 1 pile with the minimum cost.
+
+for eg:
+{3, 5, 1, 2, 6}
+we can merge {5, 1, 2} into {8}
+{3, 8, 6}
+we can merge {3, 8, 6} into {17}
+{17}
+we can stop here: our total cost = 8 + 17 = 25
+
+## The first thing we need to check can we merge all these piles into 1 pile ?
+we can check this : ## if((n-1)%(k-1) == 0) then only we can do merging.
+
+let's see how we can arrive at this formula.
+
+if there's n piles given and k piles are to be merged. Then,
```

Comment:
suggest to use `$n$` and `$k$` so that it will render as $n$ and $k$. Apply to all variables or formulas.